packages/lisk-core/src/controller.js

```diff
@@ -1,40 +1,171 @@
-'use strict';
-
-const { ModuleFactory } = require('@lisk-core/utils');
-const Bus = require('./bus');
 const Promise = require('bluebird');
+const fs = require('fs-extra');
+const config = require('./helpers/config');
+const ModuleFactory = require('./factories/modules');
+const ComponentFactory = require('./factories/components');
+const EventEmitterChannel = require('./channels/event_emitter');
+const Schema = require('./helpers/schema');
+const Bus = require('./bus');
+
+const logger = ComponentFactory.create('logger', config.components.logger);
 
 module.exports = class Controller {
 	constructor(options) {
-		console.log('Initializing controller');
+		logger.info('Initializing controller');
+		this.config = options;
 		this.modules = {};
+		this.channel = null;
 
 		// Setting up bus
-		if(!this.bus) {
+		if (!this.bus) {
 			this.bus = new Bus(this, {
 				wildcard: true,
 				delimiter: ':',
-				maxListeners: 1000
+				maxListeners: 1000,
 			});
 		}
 	}
 
-	async loadModules() {
-		const chainModule = ModuleFactory.create('child_process', 'lisk-chain', {}, this, this.bus);
-		const p2p2Module = ModuleFactory.create('in_memory', 'lisk-p2p', {}, this, this.bus);
+	async setup() {
+		await Schema.setup();
```

Comment:
Schema validation is being done once when we start the app. So we can move it to cli/index.js